### PR TITLE
Add multi-environment support for AKS

### PR DESCRIPTION
**Pull request by mjura**
Issue: https://github.com/rancher/aks-operator/issues/98

### CODE REVIEW

**Comment by kkaempf** [comment]
SURE-5437

**Comment by a-blender** [comment]
**Important**

@mjura Please _do not_ add the keyword "Fixes" to PRs but use `Issue`. There's documentation on this, and the keyword closes linked GH issues out automatically on PR merge. We don't want this because if an issue gets closed then it gets overlooked by QA instead of tested. This caused an issue to be missed and not tested during the Rancher 2.6.11 release which ended up delaying the release.

File: pkg/api/norman/customization/aks/handler.go [hunk 1]
```diff
@@ -177,20 +178,25 @@ func (h *handler) checkCredentials(req *http.Request) (int, error) {
...
+	cred.BaseURL = azureEnvironment.ResourceManagerEndpoint
+	cred.AuthBaseURL = azureEnvironment.ActiveDirectoryEndpoint
+
 	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
 	defer cancel()
 	if cred.TenantID == "" {
-		cred.TenantID, err = azureutil.FindTenantID(ctx, azure.PublicCloud, cred.SubscriptionID)
+		cred.TenantID, err = azureutil.FindTenantID(ctx, azureEnvironment, cred.SubscriptionID)
 		if err != nil {
 			return http.StatusBadRequest, fmt.Errorf("could not find tenant ID: %w", err)
```
Comment (a-blender) [inline pkg/api/norman/customization/aks/handler.go:196]: Suggestion

`return http.StatusBadRequest, fmt.Errorf("could not find tenant ID for Azure environment %w: %w", azureEnvironment, err)`

**Review by a-blender — APPROVED** [review]

File: pkg/api/norman/customization/aks/handler.go (hunk 1 above)
Comment (mjura) [reply pkg/api/norman/customization/aks/handler.go]: great suggestion, thank you

**Review by a-blender — COMMENTED** [review]
@mjura Looks like tests are failing

```
/bin/golangci-lint
105s
693	Running: golangci-lint
105s
694	pkg/api/norman/customization/aks/handler.go:196:34: printf: fmt.Errorf format %w has arg azureEnvironment of wrong type github.com/Azure/go-autorest/autorest/azure.Environment (govet)
317s
695				return http.StatusBadRequest, fmt.Errorf("could not find tenant ID for Azure environment %w: %w", azureEnvironment, err)
317s
696				                              ^
317s
697	time="2023-03-30T15:03:15Z" level=fatal msg="exit status 1"
```

**Comment by mjura** [comment]
> @mjura Looks like tests are failing
> 
 I fixed tests, please review

**Review by furkatgofurov7 — APPROVED** [review]
LGTM

**Comment by kkaempf** [comment]
tentatively adding to Q2 milestone

@mjura @a-blender who's going to merge this PR ?

@mjura shouldn't there be a respective PR for HEAD ?

**Comment by mjura** [comment]
> tentatively adding to Q2 milestone
> 
> @mjura @a-blender who's going to merge this PR ?

We are waiting for CI first 

> @mjura shouldn't there be a respective PR for HEAD ?

It should be merged to dev-v2.7

**Review by alexander-demicev — APPROVED** [review]

**Comment by a-blender** [comment]
@mjura I just had the same problem with my unit test PR. provisioning rke2/k3s tests can't be found because that script doesn't exist anymore. Rebase

**Comment by mjura** [comment]
rebased

**Pull request merged**